airflow/utils/task_group.py

```diff
@@ -653,15 +653,21 @@ def task_group_to_dict(task_item_or_group):
     """
     from airflow.models.abstractoperator import AbstractOperator
 
-    if isinstance(task_item_or_group, AbstractOperator):
+    if isinstance(task := task_item_or_group, AbstractOperator):
+        setup_teardown_type = None
+        if task.is_setup is True:
+            setup_teardown_type = "setup"
+        elif task.is_teardown is True:
+            setup_teardown_type = "teardown"
         return {
-            "id": task_item_or_group.task_id,
+            "id": task.task_id,
             "value": {
-                "label": task_item_or_group.label,
-                "labelStyle": f"fill:{task_item_or_group.ui_fgcolor};",
-                "style": f"fill:{task_item_or_group.ui_color};",
+                "label": task.label,
+                "labelStyle": f"fill:{task.ui_fgcolor};",
+                "style": f"fill:{task.ui_color};",
                 "rx": 5,
                 "ry": 5,
+                "setupTeardownType": setup_teardown_type,
```

Comment:
@bbovenzi would you say i should only include this element if not None?

Reply:
Yeah exactly, that's consistent with the type file below.

Reply:
ok thanks, updated